src/main/java/xyz/nucleoid/dungeons/dungeons/item/armor/DgArmorItem.java

```diff
@@ -0,0 +1,148 @@
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.collection.DefaultedList;
+import xyz.nucleoid.dungeons.dungeons.item.base.DgArmor;
+import xyz.nucleoid.dungeons.dungeons.item.base.DgFlavorTextProvider;
+import xyz.nucleoid.dungeons.dungeons.util.item.DgArmorUtil;
+import xyz.nucleoid.dungeons.dungeons.util.item.DgFlavorText;
+import xyz.nucleoid.dungeons.dungeons.util.item.DgItemQuality;
+import xyz.nucleoid.dungeons.dungeons.util.item.DgItemUtil;
+import xyz.nucleoid.dungeons.dungeons.util.item.material.DgArmorMaterial;
+import xyz.nucleoid.dungeons.dungeons.util.item.material.DgMaterialComponent;
+import xyz.nucleoid.dungeons.dungeons.util.item.material.DgMaterialPicker;
+import xyz.nucleoid.plasmid.fake.FakeItem;
+
+import java.util.*;
+
+public class DgArmorItem extends Item implements FakeItem, DgArmor, DgFlavorTextProvider {
+    private final EquipmentSlot slot;
+    private final double baseToughness;
+    private final DgArmorMaterial material;
+
+    public DgArmorItem(DgArmorMaterial material, EquipmentSlot slot, double baseToughness, Settings settings) {
+        super(settings);
+        this.slot = slot;
+        this.baseToughness = baseToughness;
+        this.material = material;
+    }
+
+    @Override
+    public double getBaseToughness() {
+        return baseToughness;
+    }
+
+    public EquipmentSlot getSlot() {
+        return slot;
+    }
+
+    @Override
+    public double getKnockBackResistance() {
+        return 0;
+    }
+
+    public DgArmorMaterial getMaterial() {
+        return material;
+    }
+
+    public ItemStack createStack(DgItemQuality quality) {
+        return DgArmorUtil.initArmor(DgArmorUtil.armorBuilder(this).build(), quality);
+    }
+
+    @Override
+    public void appendStacks(ItemGroup group, DefaultedList<ItemStack> stacks) {
+        for (DgItemQuality quality : DgItemQuality.values()) {
+            if (group == quality.getItemGroup()) {
+                if (quality.ordinal() >= material.getMinQuality().ordinal() && quality.ordinal() <= material.getMaxQuality().ordinal()) {
+                    stacks.add(createStack(quality));
+                }
+
+            }
+        }
+    }
+
+    @Override
+    public Set<EquipmentSlot> getValidSlots(ItemStack stack) {
+        return ImmutableSet.of(slot);
+    }
+
+    private DgFlavorText generateFlavourText(Random random, DgItemQuality quality, DgArmorMaterial material) {
+        if (quality == null || material == null) {
+            return null;
+        }
+        List<DgFlavorText> choices = new ArrayList<>();
+
+        if (quality.ordinal() <= DgItemQuality.DUSTY.ordinal()) {
+            Collections.addAll(
+                    choices,
+                    flavorTextOf("unimpressive", 1),
```

Comment:
Are these the same as for the blades? If so, will they always be? I think some of them refer to blades which doesnt make sense for armour (not sure)

Reply:
This is gonna be fixed after I rewrite everything. In the current state, everything appears as not translated, because of the different string returned in line 134.

Reply:
Ok. So, is merge blocked on that?

Reply:
Yup, look at the todo list in the parent comment, there is everything we need for a good enough merge, but I can still quickly write prompts, just so we can have a mergable PR